demos/cli_uart/cli_uart_demo.c

```diff
@@ -0,0 +1,293 @@
+                                                size_t xWriteBufferLen,
+                                                const char * pcCommandString )
+{
+    const char * pcParameter;
+    BaseType_t lParameterStringLength, xReturn;
+    static BaseType_t lParameterNumber = 0;
+
+    /* Check the write buffer is not NULL.  NOTE - for simplicity, this example assumes the
+     * write buffer length is adequate, so does not check for buffer overflows. */
+    configASSERT( pcWriteBuffer );
+
+    if( lParameterNumber == 0 )
+    {
+        /* The first time the function is called after the command has been
+         * entered just a header string is returned. */
+        sprintf( pcWriteBuffer, "The three parameters were:\r\n" );
+
+        /* Next time the function is called the first parameter will be echoed
+         * back. */
+        lParameterNumber = 1L;
+
+        /* There is more data to be returned as no parameters have been echoed
+         * back yet. */
+        xReturn = pdPASS;
+    }
+    else
+    {
+        /* Obtain the parameter string. */
+        pcParameter = FreeRTOS_CLIGetParameter
+                      (
+            pcCommandString,        /* The command string itself. */
+            lParameterNumber,       /* Return the next parameter. */
+            &lParameterStringLength /* Store the parameter string length. */
+                      );
+
+        /* Sanity check something was returned. */
+        configASSERT( pcParameter );
+
+        /* Return the parameter string. */
+        memset( pcWriteBuffer, 0x00, xWriteBufferLen );
+        sprintf( pcWriteBuffer, "%d: ", lParameterNumber );
+        strncat( pcWriteBuffer, pcParameter, lParameterStringLength );
+        strncat( pcWriteBuffer, "\r\n", strlen( "\r\n" ) );
+
+        /* If this is the last of the three parameters then there are no more
+         * strings to return after this one. */
+        if( lParameterNumber == 3L )
+        {
+            /* If this is the last of the three parameters then there are no more
+             * strings to return after this one. */
+            xReturn = pdFALSE;
+            lParameterNumber = 0L;
+        }
+        else
+        {
+            /* There are more parameters to return after this one. */
+            xReturn = pdTRUE;
+            lParameterNumber++;
+        }
+    }
+
+    return xReturn;
+}
+
+static BaseType_t prvParameterEchoCommand( char * pcWriteBuffer,
+                                           size_t xWriteBufferLen,
+                                           const char * pcCommandString )
+{
+    const char * pcParameter;
+    BaseType_t lParameterStringLength, xReturn;
+    static BaseType_t lParameterNumber = 0;
+
+    /* Remove compile time warnings about unused parameters, and check the
+     * write buffer is not NULL.  NOTE - for simplicity, this example assumes the
+     * write buffer length is adequate, so does not check for buffer overflows. */
+    ( void ) pcCommandString;
```

Comment:
There are a couple more instances of these unused parameter casts

Reply:
Fixed